src/market/config.ts

```diff
@@ -19,34 +19,51 @@ const DEFAULTS = {
  */
 export class DemandConfig {
   public readonly yagnaOptions?: YagnaOptions;
-  public readonly timeout: number;
   public readonly expiration: number;
   public readonly subnetTag: string;
   public readonly maxOfferEvents: number;
   public readonly offerFetchingInterval: number;
   public readonly logger?: Logger;
   public readonly eventTarget?: EventTarget;
+  public readonly debitNotesAcceptanceTimeoutSec: number;
+  public readonly midAgreementPaymentTimeoutSec: number;
 
   constructor(options?: DemandOptions) {
     this.subnetTag = options?.subnetTag || EnvUtils.getYagnaSubnet() || DEFAULTS.subnetTag;
-    this.timeout = options?.marketTimeout || DEFAULTS.marketTimeout;
-    this.expiration = options?.marketOfferExpiration || DEFAULTS.marketOfferExpiration;
+
+    // Handling demand/activity expiration in a backward compatible way
+    const oldExpiration = options?.marketOfferExpiration;
+    const newExpiration = options?.expiration;
+    this.expiration = oldExpiration || newExpiration || DEFAULTS.expiration;
```

Comment:
If user sets `marketOfferExpiration` or `expiration` to `0` it will be changed to `DEFAULTS.expiration`. Is this expected? Why not use `??` here?

Reply:
This value is translated to a required property of the agreement which determines the time after which the agreement will be terminated. Putting `0` here would essentially lead to a situation where demand will be rejected by all providers on the network.

Thus, I think that having a sane default here make sense, as it's going to be harder for the user to misconfigure the demand.

Reply:
What's the behavior of expiration==0?

Reply:
@pgrzy-golem , for the demand it would mean that it should expire "now", effectively rendering it useless.